### PR TITLE
Misc 42e0 funcs, new SQ macro, removing no-builtin

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/code_42e0_len_1f60.c [hunk 1]
```diff
@@ -22,45 +28,101 @@ INCLUDE_ASM("code_42e0_len_1f60", _heap_realloc);
...
+        }
+    }
+    return b - a;
+}
 
 f32 INCLUDE_ASM("code_42e0_len_1f60", atan2, f32 startX, f32 startZ, f32 endX, f32 endZ);
 
-INCLUDE_ASM("code_42e0_len_1f60", get_player_normal_yaw);
+f32 get_player_normal_yaw(void) {
+    return atan2(0, 0, (*gGameStatusPtr)->playerTraceNormal[0], (*gGameStatusPtr)->playerTraceNormal[2]);
```
Comment (nanaian) [inline src/code_42e0_len_1f60.c:107]: Consider turning `GameStatus::playerTraceNormal` into a Vec3f and using `.x` and `.z` here?

File: src/code_42e0_len_1f60.c [hunk 2]
```diff
@@ -22,45 +28,101 @@ INCLUDE_ASM("code_42e0_len_1f60", _heap_realloc);
...
 
 INCLUDE_ASM("code_42e0_len_1f60", func_80029934);
 
 INCLUDE_ASM("code_42e0_len_1f60", func_80029994);
 
 s32 INCLUDE_ASM("code_42e0_len_1f60", rand_int, s32 arg0);
 
-INCLUDE_API_ASM("code_42e0_len_1f60", signF);
+f32 signF(f32 val) {
+    f32 ret;
```
Comment (nanaian) [inline src/code_42e0_len_1f60.c:75]: Possibly call this `sign`

File: include/variables.h [hunk 3]
```diff
@@ -12,6 +12,7 @@ extern ActionState gPlayerActionState;
 extern PlayerAnim gPlayerAnimation;
 extern PlayerStatus gPlayerStatus;
 extern GameStatus* gGameStatusPtr[1];
+extern s32 D_80074410;
```
Comment (nanaian) [inline include/variables.h:15]: Do we not have enough info to name this symbol `gRandSeed` or something?

Comment (ethteck) [reply include/variables.h:15]: I wasn't sure if it was the seed itself or something involved in the seed's computation. but yeah I think that's a good name for now at least

**Review by nanaian — APPROVED** [review]

**Review by pixel-stuck — APPROVED** [review]

**Pull request merged**